lib/vmpooler/api/v1.rb

```diff
@@ -117,109 +148,56 @@ def mean(list)
+        if !validate_date_str(param.to_s)
+          halt 400, "Invalid date format '#{param}', must match YYYY-MM-DD."
+        end
       end
 
       from_date, to_date = Date.parse(from_param), Date.parse(to_param)
 
       if to_date < from_date
-        halt 400, 'Date range is invalid, "to" cannot come before "from".'
+        halt 400, 'Date range is invalid, \'to\' cannot come before \'from\'.'
       elsif from_date > Date.today
-        halt 400, 'Date range is invalid, "from" must be in the past.'
+        halt 400, 'Date range is invalid, \'from\' must be in the past.'
       end
 
-      # total clone time for entire duration requested.
-      min_max_clone_times = []    # min/max clone times from each day.
-      total_clones_per_day = []   # total clones from each day
-      total_clone_dur_day = []    # total clone durations from each day
-
-      # generate sequence/range for from_date to to_date (inclusive).
-      # for each date, calculate the day's clone total & average, as well
-      # as add to total/overall values.
       (from_date..to_date).each do |date|
-        me = {
-            date: date.to_s,
-            clone: {
-                duration: {
-                    average: 0,
-                    min: 0,
-                    max: 0,
-                    total: 0
-                },
-                count: {
-                    total: $redis.hlen('vmpooler__clone__' + date.to_s).to_i
-                }
-            }
+        daily = {
+          date: date.to_s,
+          clone: get_clone_metrics(date.to_s)
         }
 
-        # add to daily clone count array
-        total_clones_per_day.push(me[:clone][:count][:total])
-
-        # fetch clone times from redis for this date. convert the results
-        # to float (numeric).
-        clone_times = $redis.hvals('vmpooler__clone__' + date.to_s).map(&:to_f)
-
-        unless clone_times.nil? or clone_times.length == 0
-          clone_time = clone_times.reduce(:+).to_f
-          # add clone time for this date for later processing
-          total_clone_dur_day.push(clone_time)
-
-          me[:clone][:duration][:total] = clone_time
+        result[:daily].push(daily)
+      end
 
-          # min and max clone durations for the day.
-          mi, ma = clone_times.minmax
-          me[:clone][:duration][:min], me[:clone][:duration][:max] = mi, ma
-          min_max_clone_times.push(mi, ma)
+      daily_clone_counts = []
+      daily_clone_durations = []
 
-          # if clone_total > 0, calculate clone_average.
-          # this prevents divide by 0 problems.
-          if me[:clone][:count][:total] > 0
-            me[:clone][:duration][:average] = mean(clone_times)
-          else
-            me[:clone][:duration][:average] = 0
-          end
-        end
+      result[:daily].each do |daily|
+        daily_clone_counts.push(daily[:clone][:count][:total])
+        daily_clone_durations.push(daily[:clone][:duration][:min])
```

Comment:
If this `daily` did not clone anything (`[:count][:total] == 0`), then `min` and `max` will be the default of `0`. Appending those values to `daily_clone_durations` will cause future processing to select `0` as the overall minimum duration for this summary. These values should only be appended if the `daily` cloned something.